### ISSUE TITLE
Image tests: Fedora IoT tests and  RHEL 8 edge+tar image tests are failing

### ISSUE THREAD

**Issue by jrusz** [osbuild/osbuild-composer#1464]
It seems that after #1431 was merged, some of the image tests started failing. 
Namely `fedora_33-x86_64-fedora_iot_commit-boot.json` and `fedora_33-x86_64-fedora_iot_commit_debug-boot.json` which can be seen here:  https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1334367653 

It's possible the same goes for other Fedora distros but I haven't checked them.

I can see more failures for RHEL-8 here: https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1334367654
Those failures are namely: `rhel_8-x86_64-rhel_edge_commit-boot.json` `rhel_8-x86_64-rhel_edge_commit_rt-boot.json` and `rhel_8-x86_64-tar-boot.json`

The RHEL failures all seem to be the same and related to SElinux? I think it's the same for Fedora but the error diff is much longer...

**Comment by thozza** [comment]
So the issue is that for ostree-based images, the `image-info` bind-mounts `/usr/etc` to `/etc`. `setfiles` founds out that there is an association between files in `/usr/etc` and `/etc` and also finds conflicting specifications for these locations. The choice of the specification is not deterministic, thus can randomly produce different results at any time. The content of `/etc` in this case will always have incorrect SELinux labels for the location.

The ideal solution would be to analyze the SELinux ctx mismatches without the bind-mounted `/etc`. However for our use case, I propose much simpler fix/workaround - exclude `/etc` from the SELinux ctx mismatch analysis of ostree-based images. (FYI @gicmo @achilleas-k).

My proposed fix would be https://github.com/thozza/osbuild-composer/commit/8252153d5150b3d1caa93e22bf71d70ea4b240d4, which is currently being tested in https://github.com/osbuild/osbuild-composer/pull/1468

**Comment by thozza** [comment]
Update: I think that I found cause of the remaining differences in the reported `image-info` output in the CI and when regenerating image test case in a VM. The reason seems to be that some labels are not known to the host when the image test case is generated inside a VM, while they are known to the host when running the image test cases in CI. There is at least the difference, that in the CI environment, the package `osbuild-composer-tests` is installed on the system, which pulls in additional packages, some of them shipping SELinux policy. Example is `container-selinux`. I'm now verifying various scenarios and will be making additional changes to the generator script and regenerating relevant test cases.

**Comment by thozza** [comment]
After some thinking and testing, I think that rather than installing additional `-selinux` packages in the environment where the image test cases are generated, we should fix the situation that `image-info` is not able to read unknown SELinux labels. The reason is that we may be missing some of these `-selinux` packages which are used inside the image and get "false" SELinux ctx mismatches reported for the image in the image test case as well as in CI when testing it.

I think that the best solution would be to make `image-info` able to read SELinux labels which are not known to the host. From my testing, it was enough to label the `tools/image-info` with `osbuild_exec_t` type to get all labels from the image properly reported. So I would probably propose to either reuse the `osbuild_exec_t` type for `image-info` or define our own type for it. Image test cases run via `osbuild-image-tests` in CI execute the installed `/usr/libexec/osbuild-composer/image-info`, so labeling it with `osbuild_exec_t` type should be doable.

As for generating image test cases, there are two situations:

- Test case is generated locally using `tools/test-case-generators/generate-test-cases`, which calls `tools/image-info` from the git checkout. One would have to label the `tools/image-info` manually with `osbuild_exec_t` type to get correct report from it.
- Test cases are generated in a VM using `tools/test-case-generators/generate-all-test-cases`, which just calls the `tools/test-case-generators/generate-test-cases` inside the VM. However, the script could label the `tools/image-info` inside the VM with `osbuild_exec_t` before generating the test case, so it could work correctly out of the box.

**Comment by teg** [comment]
What do you think of this @gicmo? I'm surprised this works to be honest, thought we'd have to get the whole chain of binaries from PID1 to `image-info` labelled correctly (or rules adjusted). If it happens to "just work" that's great, of course.

**Comment by gicmo** [comment]
@thozza  and I had a meeting about it and I think this is the best we can do without a re-design of how we get the labels for tar and osbuild;)
About the chain, my guess is that everything up to `image-info` is not labeled and runs in `unconfined`? Would be interesting to check.

**Comment by teg** [comment]
Ah, yeah, that would make sense.

**Comment by thozza** [comment]
@gicmo is correct with the context of the running process. Relabeling just `image-info` with labels used by `osbuild` does the trick. I made necessary (and a bit hackish) changes to `tools/test-case-generators/generate-all-test-cases` script and I'm now regenerating all test cases. This will take some non-trivial time (probably till tomorrow).

I'll update PR #1468 once I have all the test cases regenerated to make sure it works. But I tested all pieces separately so there should be no surprises. Once it passes, I'll close the PR and create a new one just with my changes (the #1468 contains changes from @jrusz on top of mine).
